### PR TITLE
Prefix internal Test type variants so node-test-runner can find them

### PR DESCRIPTION
**Pull request by lydell**
**Explanation for why node-test-runner needs this:** https://github.com/rtfeldman/node-test-runner/pull/442#issuecomment-713526271

(Original discussion leading to this PR: https://github.com/rtfeldman/node-test-runner/pull/442#discussion_r495579878)

### CODE REVIEW

**Comment by harrysarson** [comment]
I really want this change, it will future proof a hack we need to make in the test runner to detect tests without parsing the .elmi files.
I think the ugliness is worth it.

**Comment by drathier** [comment]
Yes, I want more context and documentation first! https://github.com/rtfeldman/node-test-runner/pull/442#issuecomment-713526271

File: src/Test/Internal.elm [hunk 1]
```diff
@@ -7,20 +7,24 @@ import Test.Expectation exposing (Expectation(..))
 import Test.Runner.Failure exposing (InvalidReason(..), Reason(..))
 
 
+{-| All variants of this type has the `ElmTestVariant__` prefix so that
+node-test-runner can recognize them in the compiled JavaScript. This lets us
+add more variants here without having to update the runner.
+-}
```
Comment (drathier) [inline src/Test/Internal.elm:13]: Let's add a link to this PR as well so people looking at the Test type definition know what's up! https://github.com/elm-explorations/test/pull/153

**Comment by drathier** [comment]
Just want that link, then I'd be happy to merge this.

File: src/Test/Internal.elm (hunk 1 above)
Comment (lydell) [reply src/Test/Internal.elm:15]: Link added!

Comment (drathier) [reply src/Test/Internal.elm:15]: One more thing; CI wants you to elm-format it too, so CI goes green for non-linux builds :)

Comment (lydell) [reply src/Test/Internal.elm:16]: 🤦‍♂️ Fixed

Comment (drathier) [reply src/Test/Internal.elm:16]: I'll merge it when it goes green :)

**Comment by drathier** [comment]
Cool, let's give it a go :)

**Pull request merged**